mmdet/models/dense_heads/yolo_head.py

```diff
@@ -0,0 +1,581 @@
+            else:
+                pos_weight = 1
+                conf_loss_weight = 1
+
+            pos_weight = gt_label.new_tensor(pos_weight)
+
+            losses_cls = F.binary_cross_entropy_with_logits(
+                pred_label, gt_label, reduction='none')
+
+            losses_cls *= pos_mask
+
+            losses_conf = F.binary_cross_entropy_with_logits(
+                pred_conf, gt_conf, reduction='none',
+                pos_weight=pos_weight) * pos_and_neg_mask * conf_loss_weight
+
+            if xy_use_logit:
+                losses_xy = F.mse_loss(
+                    pred_t_xy, gt_t_xy, reduction='none') * pos_mask * 2
+            else:
+                losses_xy = F.binary_cross_entropy_with_logits(
+                    pred_t_xy, gt_t_xy, reduction='none') * pos_mask * 2
+
+            losses_wh = F.mse_loss(
+                pred_t_wh, gt_t_wh, reduction='none') * pos_mask * 2
+
+            losses['cls'] += torch.sum(losses_cls)
+            losses['conf'] += torch.sum(losses_conf)
+            losses['xy'] += torch.sum(losses_xy)
+            losses['wh'] += torch.sum(losses_wh)
+
+        return losses
+
+    def _preprocess_target_single_img(self,
+                                      gt_bboxes,
+                                      gt_labels,
+                                      anchor_grids,
+                                      ignore_thresh,
+                                      one_hot_smoother=0,
+                                      xy_use_logit=False):
+        """Generate matching bounding box prior and converted GT."""
+        negative_mask_across_scale = []
+        gt_t_across_scale = []
+
+        for anchor_grid in anchor_grids:  # len(anchor_grids) == num_scales
+            negative_mask_size = list(anchor_grid.size())[:-1]
+            negative_mask = anchor_grid.new_ones(
+                negative_mask_size, dtype=torch.uint8)
+            negative_mask_across_scale.append(negative_mask)
+            gt_t_size = negative_mask_size + [self.num_attrib]
+            gt_t = anchor_grid.new_zeros(gt_t_size)
+            gt_t_across_scale.append(gt_t)
+
+        for gt_bbox, gt_label in zip(gt_bboxes, gt_labels):
+            # For each gt, convert to cxywh
+            gt_cx = (gt_bbox[0] + gt_bbox[2]) / 2
+            gt_cy = (gt_bbox[1] + gt_bbox[3]) / 2
+            gt_w = gt_bbox[2] - gt_bbox[0]
+            gt_h = gt_bbox[3] - gt_bbox[1]
+            gt_bbox_cxywh = torch.stack((gt_cx, gt_cy, gt_w, gt_h))
+
+            iou_to_match_across_scale = []
+            grid_coord_across_scale = []
+
+            for i_scale in range(self.num_scales):
+                # For each sacle of each gt:
+                stride = self.strides[i_scale]
+                anchor_grid = anchor_grids[i_scale]
+                # 1. Calculate the iou between each anchor and the gt
+                iou_gt_anchor = iou_multiple_to_one(
+                    anchor_grid, gt_bbox_cxywh, center=True)
```

Comment:
Will `bbox_overlaps` do the trick?

Reply:
Yes, I think. I'll try it. Thanks.

Reply:
Sorry, I have tried but I'm afraid that using `bbox_overlaps` will involve too much work. I have updated the doc so we will see that `anchor_grid` is actually a tensor of shape `(num_anchors, num_grid_h, num_grid_w, 4)`, instead of `(N, 4)` that `bbox_overlaps` accepts. I don't think it's elegant to reshape the tensors back and forth, since there are already a lot~  -:)